pkg/gameserverallocations/metrics.go

```diff
@@ -82,8 +101,8 @@ func (r *metrics) setStatus(status string) {
 }
 
 // setError set the latency status tag as error.
-func (r *metrics) setError() {
-	r.mutate(tag.Update(keyStatus, "error"))
+func (r *metrics) setError(errorType string) {
```

Comment:
This whole function seems... odd to me.

Any reason we aren't using `stats.RecordWithTags()` everywhere instead?

We have some nice helpers too: https://github.com/googleforgames/agones/blob/96c3d26857deebd05917260501f7ac74094ba461/pkg/metrics/util.go#L45-L49

Reply:
Mmmm i'm not entirely sure why we aren't using `stats.RecordWithTags()`. There's only one other metric in this file which is the latency one and it uses `stats.Record(r.ctx, gameServerAllocationsLatency.M(time.Since(r.start).Seconds()))` directly. I could update this file and change it in this PR to use `stats.RecordWithTags()`

Reply:
SGTM

Reply:
Ohhh i think i see why, it first records the metrics and then it does a lot of manipulation with the tags for example: [latency.setRequest()](https://github.com/googleforgames/agones/blob/049de2fbbe8d548e68a79e548dd0d05bb37d8636/pkg/gameserverallocations/allocator.go#L200) mutates [tags](https://github.com/googleforgames/agones/blob/049de2fbbe8d548e68a79e548dd0d05bb37d8636/pkg/gameserverallocations/metrics.go#L90) same case with [latency.setResponse(out) ](https://github.com/googleforgames/agones/blob/049de2fbbe8d548e68a79e548dd0d05bb37d8636/pkg/gameserverallocations/allocator.go#L235C10-L235C21) I can probably merge latency.record() and latency.setRequest(gsa) but not sure about latency.setResponse(out) since that happens later in the function and the code can return before that. Wdyt ?